api/base/exceptions.py

```diff
@@ -17,43 +53,28 @@ def json_api_exception_handler(exc, context):
 
     # Import inside method to avoid errors when the OSF is loaded without Django
     from rest_framework.views import exception_handler
+
     response = exception_handler(exc, context)
 
-    # Error objects may have the following members. Title and id removed to avoid clash with "title" and "id" field errors.
-    top_level_error_keys = ['links', 'status', 'code', 'detail', 'source', 'meta']
-    resource_object_identifiers = ['type', 'id']
     errors = []
 
     if response:
         message = response.data
 
         if isinstance(exc, JSONAPIException):
-            errors.extend([
-                {
-                    'source': exc.source or {},
-                    'detail': exc.detail,
-                }
-            ])
+            errors.extend([{'source': exc.source or {}, 'detail': exc.detail}])
         elif isinstance(message, dict):
-            for error_key, error_description in message.iteritems():
-                if error_key in top_level_error_keys:
-                    errors.append({error_key: error_description})
-                elif error_key in resource_object_identifiers:
-                    if isinstance(error_description, basestring):
-                        error_description = [error_description]
-                    errors.extend([{'source': {'pointer': '/data/' + error_key}, 'detail': reason} for reason in error_description])
-                elif error_key == 'attributes':
-                    if isinstance(error_description, list):
-                        errors.extend([{'source': {'pointer': '/data/' + error_key}, 'detail': reason} for reason in error_description])
-                else:
-                    if isinstance(error_description, basestring):
-                        error_description = [error_description]
-                    errors.extend([{'source': {'pointer': '/data/attributes/' + error_key}, 'detail': reason} for reason in error_description])
-
+            errors.extend(dict_error_formatting(message, None))
         else:
             if isinstance(message, basestring):
                 message = [message]
-            errors.extend([{'detail': error} for error in message])
+            index = 0
```

Comment:
You can use `enumerate` instead of maintaining this index yourself:


```python
            for index, error in enumerate(message):
                if isinstance(error, dict):
                    errors.extend(dict_error_formatting(error, index))
                else:
                    errors.append({'detail': error})
```